src/services/healthcareProfessionalService.ts

```diff
@@ -1,132 +1,252 @@
+
+        const updatedProfessionalResult = await getHealthcareProfessionalById(professionalRef.id)
 
-        addHealthcareProfessionalResult.data = newHealthcareProfessional
+        // if we didn't get it back or have errors, this is an actual error.
+        if (updatedProfessionalResult.hasErrors || !updatedProfessionalResult.data) {
+            throw new Error(`ERROR: Error updating healthcare professional: ${JSON.stringify(updatedProfessionalResult.errors)}`)
+        }
 
-        return addHealthcareProfessionalResult
+        return {
+            data: updatedProfessionalResult.data,
+            hasErrors: false
+        }
     } catch (error) {
-        throw new Error(`Error adding healthcare professional to Facility: ${error}`)
+        console.log(`ERROR: Error updating healthcareProfessional ${id}: ${error}`)
+
+        return {
+            data: {} as gqlTypes.HealthcareProfessional,
+            hasErrors: true,
+            errors: [{
+                field: 'updateHealthcareProfessional',
+                errorCode: ErrorCode.INTERNAL_SERVER_ERROR,
+                httpStatus: 500
+            }]
+        }
     }
 }
 
-function convertToDbHealthcareProfessional(
-    id: string, healthcareProfessionalInput: gqlTypes.HealthcareProfessionalInput
-) {
+function mapGqlEntityToDbEntity(newHealthcareProfessionalId: string, input: gqlTypes.CreateHealthcareProfessionalInput) {
     return {
-        id: id, 
-        acceptedInsurance: validateInsurance(healthcareProfessionalInput.acceptedInsurance as gqlTypes.Insurance[]),
-        degrees: mapAndValidateDegrees(healthcareProfessionalInput.degrees as dbSchema.Degree[]),
-        names: mapAndValidateNames(healthcareProfessionalInput.names as dbSchema.LocaleName[]),
-        specialties: mapAndValidateSpecialties(healthcareProfessionalInput.specialties as dbSchema.Specialty[]),
-        spokenLanguages: mapAndValidateLanguages(
-            healthcareProfessionalInput.spokenLanguages as dbSchema.SpokenLanguage[]
-        ),
-        isDeleted: false,
+        id: newHealthcareProfessionalId,
+        acceptedInsurance: input.acceptedInsurance as gqlTypes.Insurance[],
+        degrees: input.degrees as dbSchema.Degree[],
+        names: input.names as dbSchema.LocalizedName[],
+        specialties: input.specialties as dbSchema.Specialty[],
+        spokenLanguages: input.spokenLanguages as gqlTypes.Locale[],
+        facilityIds: input.facilityIds ?? [] as string[],
+        //business rule: createdDate cannot be set by the user.
         createdDate: new Date().toISOString(),
+        //business rule: updatedDate is updated on every change.
         updatedDate: new Date().toISOString()
-    } as dbSchema.HealthcareProfessional
+    } satisfies dbSchema.HealthcareProfessional
 }
 
-function mapDbEntityTogqlEntity(dbEntity : firebase.DocumentData) {
+//TODO solve this later
```

Comment:
Please create an issue for this and remove the commented out code to it for now.